src/Container.js

```diff
@@ -177,49 +190,54 @@ export default class Container {
             instance,
             resolver;
         context = this._resolverContext.beginResolve(name);
+
         try {
-            if (registration.dependencyList !== undefined) {
-                for (let i = 0, len = registration.dependencyList.length; i < len; i++) {
-                    dependencyKey = registration.dependencyList[i];
-                    if (utils.isString(dependencyKey)) {
-                        if(this.isGroupRegistered(dependencyKey)) {
-                            dependency = this.resolveGroup(dependencyKey);
+            if (typeof registration.factory === "function") {
```

Comment:
I think we can do this without a change to this function. We can extend the containers resolve functionality by adding a new resolver type.

For example, in `_createDefaultResolvers()` you could add this:

```
            // A resolver that invokes an external factory to resolve the
            externalFactory: {
                resolve: (container, dependencyKey) => {
                    let args = [].slice.call(arguments);
                    args.unshift(dependencyKey.key);
                    return dependencyKey.factory.apply(container, args);
                }
            }
```
(Note the above passes additional args to the factory so if you do can pass additional arguments to the instance being resolved: `container.resolve('key', { /*additional arg 1*/})`)

Your register factory method would then become:

```
    registerFactory(name, factory) {
        this._throwIfDisposed();
        Guard.isNonEmptyString(name, 'Error calling registerFactory(name, factory). The name argument must be a string and can not be \'\'');
        Guard.isFunction(factory, `Error calling registerFactory(name, factory). Provided factory for [${name}] must be a function`);
        let dependencyKey = {
            resolver: 'externalFactory',
            factory : factory,
            isResolverKey: true
        };
        return this.register(name, dependencyKey);
    }
```

I think this would avoid some code branching in the resolve function and keep in line with some other behaviour that affects resolution.